# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/tests/browser_lazyload_telemetry.js
# Commit: 1cc86398ced0
# Full Hash: 1cc86398ced034fafc66d94799f91221420e1229
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2021-02-18 04:16:06
# Regressor Bug: 1692350
# File Overlap Count: 1
# Description:
#   Bug 1692350 - Implement a telemetry probe to collect page load data for documents that use lazyload r=emilio,chutten
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104912
# ==============================================================================

diff -r 1ca4c3365fae -r 1cc86398ced0 layout/base/tests/browser_lazyload_telemetry.js
--- a/layout/base/tests/browser_lazyload_telemetry.js	Wed Feb 17 22:02:17 2021 +0000
+++ b/layout/base/tests/browser_lazyload_telemetry.js	Wed Feb 17 22:25:02 2021 +0000
@@ -9,6 +9,12 @@
   "resource://testing-common/TelemetryTestUtils.jsm"
 );
 
+function pageLoadIsReported() {
+  const snapshot = Services.telemetry.getSnapshotForHistograms("main", false)
+    .parent;
+  return snapshot.FX_LAZYLOAD_IMAGE_PAGE_LOAD_MS;
+}
+
 function dataIsReported() {
   const snapshot = Services.telemetry.getSnapshotForHistograms("main", false)
     .content;
@@ -25,6 +31,7 @@
     .getHistogramById("LAZYLOAD_IMAGE_VIEWPORT_LOADING")
     .clear();
   Services.telemetry.getHistogramById("LAZYLOAD_IMAGE_VIEWPORT_LOADED").clear();
+  Services.telemetry.getHistogramById("FX_LAZYLOAD_IMAGE_PAGE_LOAD_MS").clear();
 
   const testTab = await BrowserTestUtils.openNewForegroundTab(
     gBrowser,
@@ -47,20 +54,21 @@
     }
   );
 
+  await BrowserTestUtils.waitForCondition(pageLoadIsReported);
+
   gBrowser.removeTab(testTab);
 
   await BrowserTestUtils.waitForCondition(dataIsReported);
 
-  const snapshot = Services.telemetry.getSnapshotForHistograms("main", false)
-    .content;
+  const snapshot = Services.telemetry.getSnapshotForHistograms("main", false);
 
   // Ensures we have 4 lazyload images.
-  is(snapshot.LAZYLOAD_IMAGE_TOTAL.values[4], 1, "total images");
+  is(snapshot.content.LAZYLOAD_IMAGE_TOTAL.values[4], 1, "total images");
   // All 4 images should be lazy-loaded.
-  is(snapshot.LAZYLOAD_IMAGE_STARTED.values[4], 1, "started to load");
+  is(snapshot.content.LAZYLOAD_IMAGE_STARTED.values[4], 1, "started to load");
   // The last image didn't reach to the viewport.
   is(
-    snapshot.LAZYLOAD_IMAGE_NOT_VIEWPORT.values[1],
+    snapshot.content.LAZYLOAD_IMAGE_NOT_VIEWPORT.values[1],
     1,
     "images didn't reach viewport"
   );
@@ -68,9 +76,13 @@
   // should be three. This includes all images except
   // the last one.
   is(
-    snapshot.LAZYLOAD_IMAGE_VIEWPORT_LOADING.sum +
-      snapshot.LAZYLOAD_IMAGE_VIEWPORT_LOADED.sum,
+    snapshot.content.LAZYLOAD_IMAGE_VIEWPORT_LOADING.sum +
+      snapshot.content.LAZYLOAD_IMAGE_VIEWPORT_LOADED.sum,
     3,
     "images reached viewport"
   );
+  ok(
+    snapshot.parent.FX_LAZYLOAD_IMAGE_PAGE_LOAD_MS.sum > 0,
+    "lazyload image page load telemetry"
+  );
 });